src/jira/client/jira-client.ts

```diff
@@ -21,9 +21,18 @@ import { uniq } from "lodash";
 import { getCloudOrServerFromGitHubAppId } from "utils/get-cloud-or-server";
 import { TransformedRepositoryId, transformRepositoryId } from "~/src/transforms/transform-repository-id";
 import { getDeploymentDebugInfo } from "./jira-client-deployment-helper";
+import { booleanFlag, BooleanFlags } from "config/feature-flags";
 
 // Max number of issue keys we can pass to the Jira API
-export const ISSUE_KEY_API_LIMIT = 500;
+const getIssueKeyLimit = async () => {
+	let ISSUE_KEY_API_LIMIT = 100;
+	if (await booleanFlag(BooleanFlags.ISSUE_KEY_API_LIMIT_IS_500)) {
+		ISSUE_KEY_API_LIMIT = 500;
+	}
+
+	return Promise.resolve(ISSUE_KEY_API_LIMIT);
```

Comment:
/nit
Just fyi `Promise.resolve` is not needed as `async` will auto wrap it with a `Promise`.